app/models/ticket.rb

```
@@ -1,183 +1,183 @@
+        Rails.logger.debug "Moved to next business day: #{current_time}"
+      end
+    end
+
+    if remaining_minutes > 0
+      Rails.logger.warn "SLA calculation exceeded #{max_days} days for Ticket ##{id}, remaining: #{remaining_minutes} minutes"
+      current_time += remaining_minutes.minutes # Fallback for remaining time
+    end
+
+    Rails.logger.debug "Due date calculated: #{current_time}"
+    current_time
+  end
```

RuboCop (1.73)

[convention] Assignment Branch Condition size for calculate_due_date is too high. [<24, 60, 16> 66.57/23]

(Metrics/AbcSize)